File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        if(s.size() != t.size())
            return false;
        std::unordered_map<char,int> check;
        
        for(int i = 0; i < s.size(); i++) {
            
            check[s[i]]++;
            check[t[i]]--;
        }
        
        
        for(auto&&[key,value] : check) {
            
            if(value > 0)
                return false;
        }
        
        
        
            
        return true; 
        
    }
};

File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if not(len(s) == len(t)):
            return False
        
        check_s = {}
        check_t = {}
        
        for x in s:
            check_s[x] = check_s.get(x,0)+1
        for y in t:
            check_t[y] = check_t.get(y,0)+1
            
        return check_s == check_t